modules/xmpp/JingleSessionPC.js

```diff
@@ -19,1476 +19,1561 @@ import * as JingleSessionState from "./JingleSessionState";
-            var protocol = candidate.protocol;
-            if (typeof protocol === 'string') {
-                protocol = protocol.toLowerCase();
-                if (protocol === 'tcp' || protocol ==='ssltcp') {
-                    if (self.webrtcIceTcpDisable)
-                        return;
-                } else if (protocol == 'udp') {
-                    if (self.webrtcIceUdpDisable)
-                        return;
+        this.peerconnection.onicecandidate = function (ev) {
+            if (!ev) {
+                // There was an incomplete check for ev before which left
+                // the last line of the function unprotected from a potential
+                // throw of an exception. Consequently, it may be argued that
+                // the check is unnecessary. Anyway, I'm leaving it and making
+                // the check complete.
+                return;
+            }
+            // XXX this is broken, candidate is not parsed.
+            const candidate = ev.candidate;
+            if (candidate) {
+                // Discard candidates of disabled protocols.
+                let protocol = candidate.protocol;
+                if (typeof protocol === 'string') {
+                    protocol = protocol.toLowerCase();
+                    if (protocol === 'tcp' || protocol === 'ssltcp') {
+                        if (self.webrtcIceTcpDisable)
+                            return;
+                    } else if (protocol == 'udp') {
+                        if (self.webrtcIceUdpDisable)
+                            return;
+                    }
                 }
             }
-        }
-        self.sendIceCandidate(candidate);
-    };
-    // Note there is a change in the spec about closed:
-    // This value moved into the RTCPeerConnectionState enum in the May 13, 2016
-    // draft of the specification, as it reflects the state of the
-    // RTCPeerConnection, not the signaling connection. You now detect a
-    // closed connection by checking for connectionState to be "closed" instead.
-    // I suppose at some point this will be moved to onconnectionstatechange
-    this.peerconnection.onsignalingstatechange = function () {
-        if (!(self && self.peerconnection)) return;
-        if (self.peerconnection.signalingState === 'stable') {
-            self.wasstable = true;
-        } else if (
-            (self.peerconnection.signalingState === 'closed'
+            self.sendIceCandidate(candidate);
+        };
+        // Note there is a change in the spec about closed:
+        // This value moved into the RTCPeerConnectionState enum in
+        // the May 13, 2016 draft of the specification, as it reflects the state
+        // of the RTCPeerConnection, not the signaling connection. You now
+        // detect a closed connection by checking for connectionState to be
+        // "closed" instead.
+        // I suppose at some point this will be moved to onconnectionstatechange
```

Comment:
I don't know, but I don't want to make any functional changes in this PR and removing the comment would require to change the code. I don't have enough time to investigate that now.